crates/e2e/tests/e2e/replace_order.rs

```diff
@@ -23,6 +23,135 @@ async fn local_node_try_replace_someone_else_order() {
     run_test(try_replace_someone_else_order_test).await;
 }
 
+#[tokio::test]
+#[ignore]
+async fn local_node_try_replace_active_order() {
+    run_test(try_replace_active_order_test).await;
+}
+
+async fn try_replace_active_order_test(web3: Web3) {
+    let mut onchain = OnchainComponents::deploy(web3.clone()).await;
+
+    let [solver] = onchain.make_solvers(to_wei(1)).await;
+    let [trader] = onchain.make_accounts(to_wei(1)).await;
+    let [token_a, token_b] = onchain
+        .deploy_tokens_with_weth_uni_v2_pools(to_wei(1_000), to_wei(1_000))
+        .await;
+
+    // Fund trader accounts
+    token_a.mint(trader.address(), to_wei(30)).await;
+
+    // Create and fund Uniswap pool
+    token_a.mint(solver.address(), to_wei(1000)).await;
+    token_b.mint(solver.address(), to_wei(1000)).await;
+    tx!(
+        solver.account(),
+        onchain
+            .contracts()
+            .uniswap_v2_factory
+            .create_pair(token_a.address(), token_b.address())
+    );
+    tx!(
+        solver.account(),
+        token_a.approve(
+            onchain.contracts().uniswap_v2_router.address(),
+            to_wei(1000)
+        )
+    );
+    tx!(
+        solver.account(),
+        token_b.approve(
+            onchain.contracts().uniswap_v2_router.address(),
+            to_wei(1000)
+        )
+    );
+    tx!(
+        solver.account(),
+        onchain.contracts().uniswap_v2_router.add_liquidity(
+            token_a.address(),
+            token_b.address(),
+            to_wei(1000),
+            to_wei(1000),
+            0_u64.into(),
+            0_u64.into(),
+            solver.address(),
+            U256::max_value(),
+        )
+    );
+
+    // Approve GPv2 for trading
+    tx!(
+        trader.account(),
+        token_a.approve(onchain.contracts().allowance, to_wei(15))
+    );
+
+    // Place Orders
+    let services = Services::new(&onchain).await;
+    services.start_protocol(solver).await;
+
+    let order = OrderCreation {
+        sell_token: token_a.address(),
+        sell_amount: to_wei(10),
+        buy_token: token_b.address(),
+        buy_amount: to_wei(5),
+        valid_to: model::time::now_in_epoch_seconds() + 300,
+        kind: OrderKind::Sell,
+        ..Default::default()
+    }
+    .sign(
+        EcdsaSigningScheme::Eip712,
+        &onchain.contracts().domain_separator,
+        SecretKeyRef::from(&SecretKey::from_slice(trader.private_key()).unwrap()),
+    );
+    let balance_before = token_a.balance_of(trader.address()).call().await.unwrap();
+    onchain.mint_block().await;
+    let order_id = services.create_order(&order).await.unwrap();
+
+    let app_data = format!(
```

Comment:
nit: Seems appropriate to move the creation of the replacement appdata closer to the creation of the replacement order.